cpp/src/arrow/engine/substrait/options.h

```diff
@@ -67,6 +69,16 @@ using NamedTableProvider =
     std::function<Result<compute::Declaration>(const std::vector<std::string>&)>;
 static NamedTableProvider kDefaultNamedTableProvider;
 
+using NamedTapKindMapper = std::function<Result<compute::Declaration>(
+    const std::string&, std::vector<compute::Declaration::Input>,
+    std::shared_ptr<compute::ExecNodeOptions>)>;
```

Comment:
This looks great.  I think we can do one more small tweak.
```suggestion
using NamedTapKindMapper = std::function<Result<compute::Declaration>(
    const std::string&, std::vector<compute::Declaration::Input>,
    std::shared_ptr<Schema>, std::string name)>;
```

Reply:
Done.